packages/json-api/src/-private/builders/query.ts

```diff
@@ -1,10 +1,63 @@
+/**
+ * @module @ember-data/json-api/request
+ */
 import { pluralize } from 'ember-inflector';
 
 import { buildBaseURL, buildQueryParams, QueryParamsSource, type QueryUrlOptions } from '@ember-data/request-utils';
 import type { ConstrainedRequestOptions, QueryRequestOptions } from '@ember-data/types/request';
 
 import { copyForwardUrlOptions, extractCacheOptions } from './-utils';
 
+/**
+ * Builds request options to query for resources, usually by a primary
+ * type, configured for the url and header expectations of most JSON:API APIs.
+ *
+ * **Basic Usage**
+ *
+ * ```ts
+ * import { query } from '@ember-data/json-api/request';
+ *
+ * const data = await store.request(query('person'));
+ * ```
+ *
+ * **With Query Params**
+ *
+ * ```ts
+ * import { query } from '@ember-data/json-api/request';
+ *
+ * const options = query('person', { include: ['pets', 'friends'] });
+ * const data = await store.request(options);
+ * ```
+ *
+ * **Supplying Options to Modify the Request Behavior**
```

Comment:
@runspired do you think we should add similar section of documentation to `findRecord`  builder?

Reply:
yep!